expr/coerce/promote.go

```diff
@@ -0,0 +1,41 @@
+package coerce
+
+import (
+	"github.com/brimdata/zed"
+)
+
+var promote = []int{
+	zed.IDInt8,       // IDUint8      = 0
+	zed.IDInt16,      // IDUint16     = 1
+	zed.IDInt32,      // IDUint32     = 2
+	zed.IDInt64,      // IDUint64     = 3
+	zed.IDInt128,     // IDUint128    = 4
+	zed.IDInt256,     // IDUint256    = 5
+	zed.IDInt8,       // IDInt8       = 6
+	zed.IDInt16,      // IDInt16      = 7
+	zed.IDInt32,      // IDInt32      = 8
+	zed.IDInt64,      // IDInt64      = 9
+	zed.IDInt128,     // IDInt128     = 10
+	zed.IDInt256,     // IDInt256     = 11
+	zed.IDInt64,      // IDDuration   = 12
+	zed.IDInt64,      // IDTime       = 13
+	zed.IDFloat16,    // IDFloat16    = 14
+	zed.IDFloat32,    // IDFloat32    = 15
+	zed.IDFloat64,    // IDFloat64    = 16
+	zed.IDFloat128,   // IDFloat64    = 17
+	zed.IDFloat256,   // IDFloat64    = 18
+	zed.IDDecimal32,  // IDDecimal32  = 19
+	zed.IDDecimal64,  // IDDecimal64  = 20
+	zed.IDDecimal128, // IDDecimal128 = 21
+	zed.IDDecimal256, // IDDecimal256 = 22
```

Comment:
You can use the constants as keys.
```suggestion
	zed.IDUint8:      zed.IDInt8,
	zed.IDUint16:     zed.IDInt16,
	zed.IDUint32:     zed.IDInt32,
	zed.IDUint64:     zed.IDInt64,
	zed.IDUint128:    zed.IDInt128,
	zed.IDUint256:    zed.IDInt256,
	zed.IDInt8:       zed.IDInt8,
	zed.IDInt16:      zed.IDInt16,
	zed.IDInt32:      zed.IDInt32,
	zed.IDInt64:      zed.IDInt64,
	zed.IDInt128:     zed.IDInt128,
	zed.IDInt256:     zed.IDInt256,
	zed.IDDuration:   zed.IDInt64,
	zed.IDTime:       zed.IDInt64,
	zed.IDFloat16:    zed.IDFloat16,
	zed.IDFloat32:    zed.IDFloat32,
	zed.IDFloat64:    zed.IDFloat64,
	zed.IDFloat128:   zed.IDFloat128,
	zed.IDFloat256:   zed.IDFloat256,
	zed.IDDecimal32:  zed.IDDecimal32,
	zed.IDDecimal64:  zed.IDDecimal64,
	zed.IDDecimal128: zed.IDDecimal128,
	zed.IDDecimal256: zed.IDDecimal256,
```

Reply:
We can do this later.

Reply:
As agreed we can fix this one later...